mlflow/models/signature.py

```diff
@@ -224,24 +224,26 @@ def infer_signature(
     Returns:
       ModelSignature
     """
-    if model_input is not None:
-        inputs = (
-            convert_dataclass_to_schema(model_input)
-            if is_dataclass(model_input)
-            else _infer_schema(model_input)
-        )
-    else:
-        inputs = None
-    if model_output is not None:
-        outputs = (
-            convert_dataclass_to_schema(model_output)
-            if is_dataclass(model_output)
-            else _infer_schema(model_output)
-        )
-    else:
-        outputs = None
-    params = _infer_param_schema(params) if params else None
-    return ModelSignature(inputs, outputs, params)
+    schemas = {"inputs": model_input, "outputs": model_output}
+    for key, data in schemas.items():
+        if data is not None:
+            try:
+                schemas[key] = (
+                    convert_dataclass_to_schema(data) if is_dataclass(data) else _infer_schema(data)
+                )
+            except Exception:
+                _logger.warning(
+                    f"Failed to infer schema for {data}. "
```

Comment:
Showing raw data makes the message hard to read. Since user has access to the input and output data, I think it's fine to just say "Failed to infer schema for outputs"?
```
2024/12/04 02:42:10 WARNING mlflow.models.signature: Failed to infer schema for {'messages': [HumanMessage(content='Hi, can you help me?', additional_kwargs={}, response_metadata={}, id='cfaab3d9-872a-448b-a95b-4fe60170b598'), AIMessage(content="Of course! I'm here to help. What do you need assistance with today?", additional_kwargs={'refusal': None}, response_metadata={'token_usage': {'completion_tokens': 18, 'prompt_tokens': 59, 'total_tokens': 77, 'completion_tokens_details': {'accepted_prediction_tokens': 0, 'audio_tokens': 0, 'reasoning_tokens': 0, 'rejected_prediction_tokens': 0}, 'prompt_tokens_details': {'audio_tokens': 0, 'cached_tokens': 0}}, 'model_name': 'gpt-3.5-turbo-0125', 'system_fingerprint': None, 'finish_reason': 'stop', 'logprobs': None}, id='run-f65783b0-60e5-463a-8dd4-a27d76bdf6eb-0', usage_metadata={'input_tokens': 59, 'output_tokens': 18, 'total_tokens': 77, 'input_token_details': {'audio': 0, 'cache_read': 0}, 'output_token_details': {'audio': 0, 'reasoning': 0}})]}. Setting schema to `Schema([ColSpec(type=AnyType())]` as default. Note that MLflow doesn't validate data types during inference for AnyType. To see the full traceback, set logging level to DEBUG.
```